presto-main/src/test/java/io/prestosql/operator/scalar/TestMathFunctions.java

```diff
@@ -1214,42 +1214,42 @@ public void testLeast()
         assertFunction("least(TINYINT'-1')", TINYINT, (byte) -1);
-        assertFunction("least(TINYINT'5', TINYINT'4', CAST(NULL AS TINYINT), TINYINT'3')", TINYINT, (byte) 3);
+        assertFunction("least(TINYINT'5', TINYINT'4', CAST(NULL AS TINYINT), TINYINT'3')", TINYINT, null);
 
         // integer
         assertFunction("least(SMALLINT'1', SMALLINT'2')", SMALLINT, (short) 1);
         assertFunction("least(SMALLINT'-1', SMALLINT'-2')", SMALLINT, (short) -2);
         assertFunction("least(SMALLINT'5', SMALLINT'4', SMALLINT'3', SMALLINT'2', SMALLINT'1', SMALLINT'2', SMALLINT'3', SMALLINT'4', SMALLINT'1', SMALLINT'5')", SMALLINT, (short) 1);
         assertFunction("least(SMALLINT'-1')", SMALLINT, (short) -1);
-        assertFunction("least(SMALLINT'5', SMALLINT'4', CAST(NULL AS SMALLINT), SMALLINT'3')", SMALLINT, (short) 3);
+        assertFunction("least(SMALLINT'5', SMALLINT'4', CAST(NULL AS SMALLINT), SMALLINT'3')", SMALLINT, null);
 
         // integer
         assertFunction("least(1, 2)", INTEGER, 1);
         assertFunction("least(-1, -2)", INTEGER, -2);
         assertFunction("least(5, 4, 3, 2, 1, 2, 3, 4, 1, 5)", INTEGER, 1);
         assertFunction("least(-1)", INTEGER, -1);
-        assertFunction("least(5, 4, CAST(NULL AS INTEGER), 3)", INTEGER, 3);
+        assertFunction("least(5, 4, CAST(NULL AS INTEGER), 3)", INTEGER, null);
 
         // bigint
         assertFunction("least(10000000000, 20000000000)", BIGINT, 10000000000L);
         assertFunction("least(-10000000000, -20000000000)", BIGINT, -20000000000L);
         assertFunction("least(50000000000, 40000000000, 30000000000, 20000000000, 50000000000)", BIGINT, 20000000000L);
         assertFunction("least(-10000000000)", BIGINT, -10000000000L);
-        assertFunction("least(500000000, 400000000, CAST(NULL as BIGINT), 300000000)", BIGINT, 300000000L);
+        assertFunction("least(500000000, 400000000, CAST(NULL as BIGINT), 300000000)", BIGINT, null);
 
         // double
         assertFunction("least(1.5E0, 2.3E0)", DOUBLE, 1.5);
         assertFunction("least(-1.5E0, -2.3E0)", DOUBLE, -2.3);
         assertFunction("least(-1.5E0, -2.3E0, -5/3)", DOUBLE, -2.3);
         assertFunction("least(1.5E0, -1.0E0 / 0.0E0, 1.0E0 / 0.0E0)", DOUBLE, Double.NEGATIVE_INFINITY);
-        assertFunction("least(5, 4, CAST(NULL as DOUBLE), 3)", DOUBLE, 3.0);
+        assertFunction("least(5, 4, CAST(NULL as DOUBLE), 3)", DOUBLE, null);
 
         // float
         assertFunction("least(REAL '1.5', 2.3E0)", DOUBLE, (double) 1.5f);
         assertFunction("least(REAL '-1.5', -2.3E0)", DOUBLE, -2.3);
         assertFunction("least(-2.3E0, REAL '-0.4', -5/3)", DOUBLE, -2.3);
         assertFunction("least(1.5E0, REAL '-1.0' / 0.0E0, 1.0E0 / 0.0E0)", DOUBLE, (double) (-1.0f / 0.0f));
-        assertFunction("least(REAL '5', 4, CAST(NULL as DOUBLE), 3)", DOUBLE, 3.0);
+        assertFunction("least(REAL '5', 4, CAST(NULL as DOUBLE), 3)", DOUBLE, null);
```

Comment:
This is a test case for DOUBLE.
Make it into a test case for REAL:

```suggestion
        assertFunction("least(REAL '5', 4, CAST(NULL AS real), 3)", REAL, null);
```

Reply:
Actually the cases above are also for DOUBLE, not REAL